### DISCUSSION TITLE
About asn1scc's license

### DISCUSSION THREAD

**Discussion by FEiseleV** [esa/asn1scc#308] [Q&A] [answered]
asn1scc itself is protected under the GPL-2.0 license. However, there is an exception notice that reads as follows:
_2. The exception is that **you may use, copy, link, modify and distribute**
**under your own terms**, binary object code versions of works based on the
tool Run-time Libraries as well as **code automatically generated by the tool**._

Do I understand this correctly that any code generated by asn1scc would not fall under the GPL-2.0 license? Meaning when I create my own ASN.1 schema, run that through the compiler to generate C source code, I can freely use that C source code?

Thank you for your assistance.

**Comment by maxime-esa** [comment, marked as answer]
Yes absolutely. The generated code is yours.
